lib/infrastructure/class-wp-rest-server.php

```diff
@@ -517,8 +551,17 @@ public function get_routes() {
 				// Single endpoint, add one deeper
 				$handlers = array( $handlers );
 			}
+			if ( ! isset( $this->route_options[ $route ] ) ) {
+				$this->route_options[ $route ] = array();
+			}
 
 			foreach ( $handlers as $key => &$handler ) {
+				if ( ! is_numeric( $key ) ) {
+					// Route option, move it to the options
+					$this->route_options[ $route ][ $key ] = $handler;
+					unset( $handler );
```

Comment:
@danielbachhuber This should help with #1262, since now you can specify "options" on the routes too. Should fix the repetition of the schema.